flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -1091,12 +1093,12 @@ private Acknowledge suspendExecution(final Exception cause) {
 		suspendAndClearExecutionGraphFields(cause);
 
 		// the slot pool stops receiving messages and clears its pooled slots
-		slotPoolGateway.suspend();
+		CompletableFuture<Acknowledge> slotPoolSuspendFuture = slotPoolGateway.suspend();
 
 		// disconnect from resource manager:
 		closeResourceManagerConnection(cause);
 
-		return Acknowledge.get();
+		return slotPoolSuspendFuture;
```

Comment:
- `slotPoolGateway.suspend` is a RPC which is executed asynchronously because `slotPool` is a `RpcEndpoint`
- I think it does not matter much in which order `slotPoolGateway.suspend()` and `closeResourceManagerConnection()` are called. In the worst case the `slotPool` will send some requests to the `ResourceManager` which will be rejected because the `JobMaster` is no longer connected to it.